SDL/Makefile

```diff
@@ -4,20 +4,31 @@
 #
 
 CCSOURCES=$(filter-out LST.cc, $(wildcard *.cc))
-CCOBJECTS=$(CCSOURCES:.cc=_$(BACKEND).o)
+CCOBJECTS_CPU=$(CCSOURCES:.cc=_cpu.o)
+CCOBJECTS_GPU=$(CCSOURCES:.cc=_gpu.o)
 
 LSTSOURCES=LST.cc
-LSTOBJECTS=$(LSTSOURCES:.cc=_cpu.o)
+LSTOBJECTS_CPU=$(LSTSOURCES:.cc=_cpu.o)
+LSTOBJECTS_GPU=$(LSTSOURCES:.cc=_gpu.o)
 
-LIB=libsdl.so
+LIB_GPU=libsdl_gpu.so
+LIB_CPU=libsdl_cpu.so
+
+ifeq ($(BACKEND), cpu)
+  LIB_GPU=
+else ifeq ($(BACKEND), gpu)
+  LIB_CPU=
+endif
+
+LIBS=$(LIB_GPU) $(LIB_CPU)
 
 #
 # flags to keep track of
 #
 
 CXX                  = g++
-CXXFLAGS             = -g -Wall -Wshadow -Woverloaded-virtual -fPIC -fopenmp -I..
-CXXFLAGS_CUDA        = -g --compiler-options -Wall --compiler-options -Wshadow --compiler-options -Woverloaded-virtual --compiler-options -fPIC --compiler-options -fopenmp -dc -lineinfo --ptxas-options=-v --cudart shared -arch=compute_70 --use_fast_math --default-stream per-thread -I..
+CXXFLAGS_CPU             = -g -Wall -Wshadow -Woverloaded-virtual -fPIC -fopenmp -I..
+CXXFLAGS_GPU        = -g --compiler-options -Wall --compiler-options -Wshadow --compiler-options -Woverloaded-virtual --compiler-options -fPIC --compiler-options -fopenmp -dc -lineinfo --ptxas-options=-v --cudart shared -arch=compute_70 --use_fast_math --default-stream per-thread -I..
```

Comment:
Yeah, that's a good point. So in general should we start calling the backend "cuda" instead of "gpu"? Or do you think it would be possible to give `sdl_run` something like `-b gpu` and depending on the hardware decide whether to use CUDA or HIP?